test/controllers/deploys_controller_test.rb

```diff
@@ -417,15 +421,59 @@ def self.with_and_without_project(&block)
     end
 
     describe "#confirm" do
+      def confirm
+        post :confirm, params: {project_id: project.to_param, stage_id: stage.to_param, deploy: {reference: "master"}}
+      end
+
+      def pull_request_stub(data_overrides = {})
+        risks = data_overrides.delete(:risks)
+
+        attrs = {
+          number: 1,
+          title: 'Cool Stuff!',
+          additions: 0,
+          deletions: 0,
+        }.merge(data_overrides)
+
+        pr = Changeset::PullRequest.new('foo/bar', Sawyer::Resource.new(Sawyer::Agent.new(''), attrs))
+        pr.stubs(:parse_risks).returns(risks)
+        pr
+      end
+
       before do
         Deploy.delete_all # triggers more callbacks
-
-        post :confirm, params: {project_id: project.to_param, stage_id: stage.to_param, deploy: {reference: "master"}}
       end
 
       it "renders the template" do
+        confirm
         assert_template :changeset
       end
+
+      it 'shows risks' do
+        pr = pull_request_stub(risks: 'The sky is falling!')
+        Deploy.any_instance.stubs(:changeset).returns(changeset(pull_requests: [pr]))
+
+        confirm
+
+        assert_response :success
+
+        assert_select 'h5' do
+          assert_select 'strong', text: '#1'
+          assert_select 'a[href=?]', 'https://github.com/foo/bar/pull/1', text: 'Cool Stuff!'
+        end
+        assert_select 'p', text: 'The sky is falling!'
```

Comment:
tests that break when the markup changes ... not a fan ... assert that it includes `Cool Stuff` and the url ?